test/test_unistd.rs

```diff
@@ -104,6 +104,74 @@ mod linux_android {
     }
 }
 
+#[test]
+// `getgroups()` and `setgroups()` do not behave as expected on Apple platforms
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_setgroups() {
+    // Skip this test when not run as root as `setgroups()` requires root.
+    if !Uid::current().is_root() {
+        use std::io;
+        let stderr = io::stderr();
+        let mut handle = stderr.lock();
+        writeln!(handle, "test_setgroups requires root privileges. Skipping test.").unwrap();
+        return
+    }
+
+    #[allow(unused_variables)]
+    let m = ::GROUPS_MTX.lock().expect("Mutex got poisoned by another test");
+
+    // Save the existing groups
+    let old_groups = getgroups().unwrap();
+
+    // Set some new made up groups
+    let groups = [Gid::from_raw(123), Gid::from_raw(456)];
+    setgroups(&groups).unwrap();
+
+    let new_groups = getgroups().unwrap();
+    assert_eq!(new_groups, groups);
+
+    // Revert back to the old groups
+    setgroups(&old_groups).unwrap();
+}
+
+#[test]
+// `getgroups()` and `setgroups()` do not behave as expected on Apple platforms
+#[cfg(not(any(target_os = "ios", target_os = "macos")))]
+fn test_initgroups() {
+    // Skip this test when not run as root as `initgroups()` and `setgroups()`
+    // require root.
+    if !Uid::current().is_root() {
+        use std::io;
+        let stderr = io::stderr();
+        let mut handle = stderr.lock();
+        writeln!(handle, "test_initgroups requires root privileges. Skipping test.").unwrap();
+        return
+    }
+
+    #[allow(unused_variables)]
+    let m = ::GROUPS_MTX.lock().expect("Mutex got poisoned by another test");
+
+    // Save the existing groups
+    let old_groups = getgroups().unwrap();
+
+    // It doesn't matter if the root user is not called "root" or if a user
+    // called "root" doesn't exist. We are just checking that the extra,
+    // made-up group, `123`, is set.
+    // FIXME: This only tests half of initgroups' functionality.
```

Comment:
What's the other half and is it easy to test?

Reply:
The half being tested is that `initgroups()` sets the made-up group `123`. Whether it fetches & sets the groups that the user is a member of is untested, because to do that we'd have to control which groups a user is a part of...which I don't actually know how to do from a libc API 🤔

Reply:
Please document in that FIXME this.